### ISSUE TITLE
Shaders don't seem to be working correctly in gles2

### ISSUE THREAD

**Issue by Shadowblitz16** [godotengine/godot#43871]
**Godot version:**

3.2.3

**OS/device including version:**

Zorin OS 15.3 Nvidia GTX 650 450 proprietary drivers

**Issue description:**

Shaders do not seem to be able to both use default alpha and sample from the draw
![image](https://user-images.githubusercontent.com/2453707/100292419-4017b600-2f35-11eb-9122-5df828333cea.png)
![image](https://user-images.githubusercontent.com/2453707/100292432-4e65d200-2f35-11eb-88b1-3a9e58c43a7b.png)

**Steps to reproduce:**
- Make a sliderbar
- Attack a shader to it with the code above
- See it doesn't work 

**Minimal reproduction project:**

not needed

**Comment by Calinou** [comment]
> **Minimal reproduction project:**
>
>not needed

Please, *always* include a minimal reproduction project. Any time saved helps contributors fix bugs quicker :slightly_smiling_face:

**Comment by Shadowblitz16** [comment]
wasn't able to reproduce the 3rd picture but I updated my issue to include the project

**Comment by akien-mga** [comment]
I don't think you can write twice to `COLOR`. Use an intermediate variable and assign to `COLOR` last.

**Comment by Shadowblitz16** [comment]
but shouldn't writing just to Color.rgb keep the alpha?
